documentation/0013-node-output-to-cli/index.md

```diff
@@ -0,0 +1,191 @@
+---
+title: Output to the command line using Node
+description: 'How to print to the command line console using Node, from the basic console.log to more complex scenarios'
+author: flaviocopes
+---
+
+<!-- TOC -->
+
+- [Basic output using the console module](#basic-output-using-the-console-module)
+- [Clear the console](#clear-the-console)
+- [Counting elements](#counting-elements)
+- [Print the stack trace](#print-the-stack-trace)
+- [Calculate the time spent](#calculate-the-time-spent)
+- [stdout and stderr](#stdout-and-stderr)
+- [Color the output](#color-the-output)
+- [Create a progress bar](#create-a-progress-bar)
+
+<!-- /TOC -->
+
+## Basic output using the console module
+
+Node provides a [`console` module](https://nodejs.org/api/console.html) which provides tons of very useful ways to interact with the command line.
+
+It is basically the same as the `console` object you find in the browser.
+
+The most basic and most used method is `console.log()`, which prints the string you pass to it to the console.
+
+If you pass an object, it will render it as a string.
+
+You can pass multiple variables to `console.log`, for example:
+
+```js
+const x = 'x'
+const y = 'y'
+console.log(x, y)
+```
+
+and Node will print both.
+
+We can also format pretty phrases by passing variables and a format specifier.
+
+For example:
+
+```js
+console.log('My %s has %d years', 'cat', 2)
+```
+
+- `%s` format a variable as a string
+- `%d` or `%i` format a variable as an integer
+- `%f` format a variable as a floating point number
+- `%O` used to print an object representation
+
+Example:
+
+```js
+console.log('%O', Number)
+```
+
+## Clear the console
+
+`console.clear()` clears the console (the behavior might depend on the console used)
+
+## Counting elements
+
+`console.count()` is a handy method.
+
+Take this code:
+
+```js
+const x = 1
+const y = 2
+const z = 3
+console.count(
+  'The value of x is ' + x + ' and has been checked .. how many times?'
+)
+console.count(
+  'The value of x is ' + x + ' and has been checked .. how many times?'
+)
+console.count(
+  'The value of y is ' + y + ' and has been checked .. how many times?'
+)
+```
+
+What happens is that count will count the number of times a string is printed, and print the count next to it:
+
+You can just count apples and oranges:
+
+```js
+const oranges = ['orange', 'orange']
+const apples = ['just one apple']
+oranges.forEach(fruit => {
+  console.count(fruit)
+})
+apples.forEach(fruit => {
+  console.count(fruit)
+})
```

Comment:
I think using semicolons at the end of each statement for at least the larger code blocks is in the best interest for a beginner going through these guides. Although there are no changes in the behavior here, I think it's best for a programmer to use semicolons while writing Js Code